src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -903,4 +902,44 @@ private TextEdit[] RemapTextEditsCore(Uri virtualDocumentUri, RazorCodeDocument
 
         return edits;
     }
+
+    private static IReadOnlyList<ClassifiedSpanInternal> GetClassifiedSpans(RazorCodeDocument document)
```

Comment:
The mapping service was the only user of `syntaxTree.GetClassifiedSpans()` outside of tests, which don't have a `RazorCodeDocument` that could be used with a new extension method, and the `RazorSyntaxFactsService` which Fred recently tried to remove, but was stuck by web tools. Assuming the web tools issue is on the cards to solve, then the existing extension method can be moved to a test project directly, so I figured keeping this contained within the mapping service, and not using the existing extension method made the most sense to avoid locking in any existing tech debt.